File: ios/BigCityPlumbing/Models/HelpGuide.swift
import Foundation

struct GuideSection: Codable, Identifiable, Hashable {
    var id: String { heading }
    let heading: String
    let body: String
}

struct HelpGuide: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    let summary: String
    let icon: String
    let sections: [GuideSection]
}

private struct HelpGuidesFile: Codable {
    let guides: [HelpGuide]
}

enum HelpGuideLoader {
    /// Loads the bundled `help_guides.json` from the app bundle.
    /// Returns an empty array on any decode failure (defensive).
    static func loadAll() -> [HelpGuide] {
        guard let url = Bundle.main.url(forResource: "help_guides", withExtension: "json") else {
            assertionFailure("help_guides.json missing from app bundle")
            return []
        }
        do {
            let data = try Data(contentsOf: url)
            let decoded = try JSONDecoder().decode(HelpGuidesFile.self, from: data)
            return decoded.guides
        } catch {
            assertionFailure("Failed to decode help_guides.json: \(error)")
            return []
        }
    }

    static func find(id: String) -> HelpGuide? {
        loadAll().first { $0.id == id }
    }
}

File: android/app/src/main/java/com/bigcityplumbing/model/HelpGuide.kt
package com.bigcityplumbing.model

import android.content.Context
import com.bigcityplumbing.R
import org.json.JSONObject
import java.io.BufferedReader
import java.io.InputStreamReader

data class GuideSection(
    val heading: String,
    val body: String,
)

data class HelpGuide(
    val id: String,
    val title: String,
    val summary: String,
    val icon: String,
    val sections: List<GuideSection>,
)

object HelpGuideLoader {
    /**
     * Loads and parses the bundled help_guides.json from res/raw.
     * Returns an empty list if anything goes wrong (defensive).
     */
    fun load(context: Context): List<HelpGuide> {
        return try {
            val raw = context.resources.openRawResource(R.raw.help_guides).use { stream ->
                BufferedReader(InputStreamReader(stream, Charsets.UTF_8)).readText()
            }
            val root = JSONObject(raw)
            val arr = root.getJSONArray("guides")
            (0 until arr.length()).map { i ->
                val g = arr.getJSONObject(i)
                val sectionsArr = g.getJSONArray("sections")
                val sections = (0 until sectionsArr.length()).map { j ->
                    val s = sectionsArr.getJSONObject(j)
                    GuideSection(
                        heading = s.getString("heading"),
                        body = s.getString("body"),
                    )
                }
                HelpGuide(
                    id = g.getString("id"),
                    title = g.getString("title"),
                    summary = g.getString("summary"),
                    icon = g.optString("icon", "info"),
                    sections = sections,
                )
            }
        } catch (e: Exception) {
            emptyList()
        }
    }
}
